crypto/multisig/src/v2.rs

```diff
@@ -0,0 +1,1260 @@
+// Copyright (c) 2018-2022 The MobileCoin Foundation
+
+//! A multi-signature implementations that supports nesting.
+//! This allows us to accommodate a requirement, where for example we want to
+//! have 2-out-of-3 organizations signing a transaction, and having each
+//! organization require 2-out-of-3 members to sign.
+
+use super::{MultiSig, MAX_SIGNATURES};
+use alloc::vec::Vec;
+use core::hash::Hash;
+use mc_crypto_digestible::Digestible;
+use mc_crypto_keys::{Signature, SignatureError, Verifier};
+use prost::{Message, Oneof};
+use serde::{Deserialize, Serialize};
+
+/// A marker traint for indicating that a type is able to produce signatures.
+pub trait Signer:
```

Comment:
I find this trait a bit confusing

1. It seems like we have a lot of other Signer traits, and they usually offer functions like `.sign` or something? Can we use one of those?
2. It seems odd that this Signer trait doesn't actually offer any way to sign something
3. It seems to be intended that the signer is actually a public key. But usually our other signer traits are implemented on a private key.

Maybe this trait should actually be called `SignerPublicKey`, and the quality it has is that signatures can be verified against it (rather than that it is able to produce signatures?)

Reply:
Looks like this can be `Default + Message + PublicKey` instead of all the other traits. What it is trying to convey is a "signer identity", which in our case is a PublicKey like you said. So I could rename it (`SignerIdentity`?) and force that it is `PublicKey`, or rename and leave as is since we don't actually care if it's a public key, we care that `Verifier<SIG>` is implemented for it

Reply:
that seems good

Reply:
Just the rename? or with the limiting on `PublicKey` to make it more obvious what the intended use is?

Reply:
I'm okay with just a rename I guess, or using the `mc-crypto-keys::PublicKey`, I think both are reasonable and resolve my confusion.